server/test/repository/QuestionRepositoryTest.java

```diff
@@ -299,4 +300,201 @@ public void createOrUpdateDraft_managesUniversalTagCorrectly()
   }
+
+  @Test
+  public void createOrUpdateDraft_managesPrimaryApplicantInfoTagsCorrectl()
+      throws UnsupportedQuestionTypeException {
+    QuestionModel nameQuestion = testQuestionBank.applicantName();
+    QuestionModel dateQuestion = testQuestionBank.applicantDate();
+    QuestionModel emailQuestion = testQuestionBank.applicantEmail();
+    QuestionModel phoneQuestion = testQuestionBank.applicantPhone();
+    ImmutableSet<PrimaryApplicantInfoTag> nameTag =
+        ImmutableSet.of(PrimaryApplicantInfoTag.APPLICANT_NAME);
+    ImmutableSet<PrimaryApplicantInfoTag> dateTag =
+        ImmutableSet.of(PrimaryApplicantInfoTag.APPLICANT_DOB);
+    ImmutableSet<PrimaryApplicantInfoTag> emailTag =
+        ImmutableSet.of(PrimaryApplicantInfoTag.APPLICANT_EMAIL);
+    ImmutableSet<PrimaryApplicantInfoTag> phoneTag =
+        ImmutableSet.of(PrimaryApplicantInfoTag.APPLICANT_PHONE);
+    QuestionDefinition nameQuestionDefinition;
+    QuestionDefinition dateQuestionDefinition;
+    QuestionDefinition emailQuestionDefinition;
+    QuestionDefinition phoneQuestionDefinition;
+
+    // Create new draft, ensure tags are correct
+    nameQuestionDefinition =
+        new QuestionDefinitionBuilder(nameQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(nameTag)
+            .build();
+    dateQuestionDefinition =
+        new QuestionDefinitionBuilder(dateQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(dateTag)
+            .build();
+    emailQuestionDefinition =
+        new QuestionDefinitionBuilder(emailQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(emailTag)
+            .build();
+    phoneQuestionDefinition =
+        new QuestionDefinitionBuilder(phoneQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(phoneTag)
+            .build();
+
+    nameQuestion = repo.createOrUpdateDraft(nameQuestionDefinition);
+    dateQuestion = repo.createOrUpdateDraft(dateQuestionDefinition);
+    emailQuestion = repo.createOrUpdateDraft(emailQuestionDefinition);
+    phoneQuestion = repo.createOrUpdateDraft(phoneQuestionDefinition);
+
+    assertThat(
+            nameQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_NAME.getQuestionTag()))
+        .isTrue();
+    assertThat(
+            dateQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_DOB.getQuestionTag()))
+        .isTrue();
+    assertThat(
+            emailQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_EMAIL.getQuestionTag()))
+        .isTrue();
+    assertThat(
+            phoneQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_PHONE.getQuestionTag()))
+        .isTrue();
+
+    versionRepo.publishNewSynchronizedVersion();
+
```

Comment:
suggestion (non-blocking):
```suggestion
// Clear tags and ensure they are cleared
```